qiskit/ignis/verification/tomography/fitters/gateset_fitter.py

```diff
@@ -0,0 +1,662 @@
+            need not be physical.
+        """
+        n = len(self.gateset_basis.spam_labels)
+        m = len(self.gateset_basis.gate_labels)
+        gram_matrix = np.zeros((n, n))
+        gate_matrices = []
+        for i in range(m):
+            gate_matrices.append(np.zeros((n, n)))
+
+        for i in range(n):  # row
+            for j in range(n):  # column
+                F_i = self.gateset_basis.spam_labels[i]
+                F_j = self.gateset_basis.spam_labels[j]
+                gram_matrix[i][j] = self.probs[(F_i, F_j)]
+
+                for k in range(m):  # gate
+                    G_k = self.gateset_basis.gate_labels[k]
+                    gate_matrices[k][i][j] = self.probs[(F_i, G_k, F_j)]
+
+        gram_inverse = np.linalg.inv(gram_matrix)
+
+        gates = [gram_inverse @ gate_matrix for gate_matrix in gate_matrices]
+        return dict(zip(self.gateset_basis.gate_labels, gates))
+
+    def default_init_state(self, size):
+        if size == 4:
+            return np.array([[0.5], [0], [0], [0.5]])
+        raise RuntimeError("No default init state for more than 1 qubit")
+
+    def default_measurement_op(self, size):
+        if size == 4:
+            return np.array([[1, 0, 0, 1]])
+        raise RuntimeError("No default measurement op for more than 1 qubit")
+
+    def fit(self) -> Dict:
+        """
+        Reconstruct a gate set from measurement data using optimization.
+
+        Returns:
+           For each gate in the gateset: its approximation found using the
+           optimization process.
+
+        Additional Information:
+            The gateset optimization process con/.sists of three phases:
+            1) Use linear inversion to obtain an initial approximation.
+            2) Use gauge optimization to ensure the linear inversion results
+            are close enough to the expected optimization outcome to serve
+            as a suitable starting point
+            3) Use MLE optimization to obtain the final outcome
+        """
+        linear_inversion_results = self.linear_inversion()
+        n = len(self.gateset_basis.spam_labels)
+        E = self.default_measurement_op(n)
+        rho = self.default_init_state(n)
+        Gs = [self.gateset_basis.gate_matrices[label]
+              for label in self.gateset_basis.gate_labels]
+        Fs = [self.gateset_basis.spam_matrix(label)
+              for label in self.gateset_basis.spam_labels]
+        Gs_E = [linear_inversion_results[label]
+                for label in self.gateset_basis.gate_labels]
+        gauge_opt = GaugeOptimize(Gs, Gs_E, Fs, rho)
+        Gs_E = gauge_opt.optimize()
+        optimizer = GST_Optimize(self.gateset_basis.gate_labels,
+                                 self.gateset_basis.spam_spec,
+                                 self.probs)
+        optimizer.set_initial_value(E, rho, Gs_E)
+        optimization_results = optimizer.optimize()
+        return optimization_results
```

Comment:
As already mentioned the objects this returns in the dict should be quantum_info objects like PTM/Choi, and DensityMatrix

Reply:
Done in 9a10c52